File: Golang/strategy/quack_behavior.go
package strategy

import "fmt"

type QuackBehavior interface {
	Quack()
}

type Quack struct{}

func(q Quack) Quack() {
	fmt.Println("Quack!")
}

type Squeak struct{}

func(s Squeak) Quack() {
	fmt.Println("Squeak!")
}

type MuteQuack struct{}

func(mq MuteQuack) Quack() {
	fmt.Println("Can't Quack!")
}

File: Python/strategy/quack_behavior.py
from abc import ABC, abstractmethod


class QuackBehavior(ABC):
    @abstractmethod
    def quack(self) -> None:
        raise NotImplementedError


class Quack(QuackBehavior):
    def quack(self) -> None:
        print("Quack!")


class Squeak(QuackBehavior):
    def quack(self) -> None:
        print("Squeak!")


class MuteQuack(QuackBehavior):
    def quack(self) -> None:
        print("Can't quack!")
